tests/test_workspaces.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+/// Test adding a second workspace
+#[test]
+fn test_workspaces_add_second_workspace() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "--git", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    std::fs::write(main_path.join("file"), "contents").unwrap();
```

Comment:
Does std::fs::write create directories? What's creating env_root/main so that you can make env_root/main/file?

Reply:
The `jj init --git main` call on line 25 creates it. And no, `std::fs::write` will otherwise fail. There's `std::fs::create_dir()` and `std::fs::create_dir_all()` for `mkdir` and `mkdir -p` functionality.

Reply:
ah, of course. I had managed to miss that. :)